### PR TITLE
fix warnings

### PR DESCRIPTION
**Pull request by rileyberton**
Compile warnings under -Wall -Werror have been fixed.. mostly this is unused variables.

### CODE REVIEW

File: src/modules/histogram.c [hunk 1]
```diff
@@ -28,6 +28,7 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include <sys/int_fmtio.h>
```
Comment (aidancully) [inline src/modules/histogram.c]: shouldn't this be `#include <inttypes.h>`?

File: src/modules/lua_check.c [hunk 2]
```diff
@@ -698,11 +696,10 @@ noit_lua_module_config(noit_module_t *mod,
 
 static int
 noit_lua_module_init(noit_module_t *mod) {
-  struct module_conf *mc;
   struct module_tls_conf *mtlsc;
   LMC_DECL(L, mod, object);
 
-  mc = noit_module_get_userdata(mod);
+  noit_module_get_userdata(mod);
```
Comment (aidancully) [inline src/modules/lua_check.c]: I think this is a no-op? just remove the call?

File: src/noit_conf_checks.c [hunk 3]
```diff
@@ -41,6 +41,7 @@
 #include <libxml/parser.h>
 #include <libxml/tree.h>
 #include <libxml/xpath.h>
+#include <sys/int_fmtio.h>
```
Comment (aidancully) [inline src/noit_conf_checks.c]: probably should be `inttypes.h`

Comment (rileyberton) [inline src/noit_conf_checks.c]: Ahh yes.  Good point.

File: src/modules/lua_check.c (hunk 2 above)
Comment (rileyberton) [inline src/modules/lua_check.c]: Not sure what side-effects get_userdata has on `mod`

Comment (aidancully) [inline src/modules/lua_check.c]: If I read [this](https://github.com/circonus-labs/reconnoiter/blob/master/src/noit_module.c#L326-L334) right, none.

Comment (rileyberton) [inline src/modules/lua_check.c]: Right, will remove.

**Comment by rileyberton** [comment]
merged by hand

**Pull request closed without merging**